teps/0098-workflows.md

```diff
@@ -256,11 +306,164 @@ Why should this TEP _not_ be implemented?
+compares Pipelines as Code to the goals of the Workflows project.
+
+Suggested steps:
+1. Design and implement a solution for parameter substitution in Pipelines as Code that's Tekton conformant.
+(The current [syntax](https://pipelinesascode.com/docs/guide/authoringprs/#authoring-pipelineruns-in-tekton-directory)
+for variable expansion isn't Tekton conformant.)
+1. Move the project to the tektoncd org, rename it, run tests on Tekton infrastructure, and publish releases to the tekton-releases GCP project.
+Use tekton standards for code, conduct, and contributions. Choose and document a Tekton stability level for the existing API.
+1. Update the project to make more extensive use of remote resolution. For example, we may want to keep the ability to refer to Tekton CRDs in the same
+repo using local paths, but encourage the use of the built-in hub resolvers rather than using the Pipelines as Code
+[hub resolution](https://pipelinesascode.com/docs/guide/resolver/#tekton-hubhttpshubtektondev) and the built-in git resolver rather than using
+the Pipelines as Code [remote URL syntax](https://pipelinesascode.com/docs/guide/resolver/#remote-http-url).
+(Note: this work is already planned for Pipelines as Code.)
+1. Create a Workflow CRD that allows users to configure much of the behavior that's currently configured through annotations, such as which events
+trigger a PipelineRun and concurrency controls to apply. We may choose to rework the existing implementation so that configuring a PipelineRun
+with annotations would be implemented by creating a Workflow under the hood. We'd support the existing API (PipelineRun with annotations) for at
+least 6 months, and then we can reevaluate how to reduce duplication and create a more consistent API while still maintaining an easy setup experience
+for users who want to configure workflows only via their Git repo rather than on the Kubernetes cluster.
+1. Set a long term goal of updating the project to use Triggers. The project used to be based on Triggers (for more info, see the 
+[latest branch](https://github.com/openshift-pipelines/pipelines-as-code/tree/release-0.5.0/config) using a Triggers backend),
+but migrated to a custom implementation due to several of the [pain points](#use-of-triggers) identified above.
+We'll have to fix these pain points before reworking the implementation to use Triggers, and the refactoring may be challenging to do incrementally
+or not worth the time required to do so. We'll have to weigh the difficulty of refactoring against the difficulty of maintaining
+similar code in both projects.
+1. Explore building extensibility mechanisms into the project. It's not yet clear what this would look like.
+
+### Adopt Pipelines as Code alongside Workflows
+
+In this solution, the Tekton community would adopt and maintain the Pipelines as Code project, but not
+use it as a Workflows replacement. Instead, it could serve as an example of how to build a CI/CD platform
```

Comment:
I am not quite sure what do you mean by "an example" and it probably would confyse users too, would that be a `tekton-contrib` of some sorts?

Reply:
I meant something along the lines of promoting PAC as an example of how platform builders can build on top of Tekton Pipelines; does that make sense?

I don't think this solution is the best way forward, because it's not totally clear what purpose having PAC in the Tekton org would serve in this case, but wanted to include it since I've gotten questions about this possibility.